app/core/components/EditSupportingFileDisplay.tsx

```diff
@@ -24,7 +28,23 @@ const EditSupportingFileDisplay = ({ name, size, url, uuid, moduleId, setQueryDa
           <TrashCan
             size={24}
             className="inline-block h-6 w-6 fill-current align-middle text-red-500"
-            onClick={async () => {
+            onClick={() => setConfirmDeleteOpen(true)}
+            aria-label="Delete file"
+          />
+          <Modal
+            title="Confirm removing a supporting file"
+            body={
+              <span>
+                The supporting file, &quot;<span className="font-bold">{name}</span>&quot;, will be
+                removed.
+              </span>
+            }
+            primaryAction="Remove"
```

Comment:
```suggestion
            primaryAction="Delete File"
```

It is good to have the button describe the action. This is for accessibility purposes - clarity comes from being verbose and not implying anything.